src/main/scala/exercise2/Queue.scala

```diff
@@ -0,0 +1,21 @@
+package exercise2
+
+import java.util.EmptyStackException
+import scala.util.{Failure, Success, Try}
+
+class Queue[T](val in: StackLike[T] = new Stack[T](), val out: StackLike[T] = new Stack[T]()) extends QueueLike[T] {
```

Comment:
A trick one can use is to have the reversing of `in` in the constructor of the queue. This way it's guaranteed that the queue will have at least one element in out (if it has any elements at all) and will need less checks for that in `dequeue` and `front`.